docs/configurations.md

```diff
@@ -0,0 +1,227 @@
+| --------------------------------------------- | -------- | ----------------------------------------------------------------------------------------- |
+| compatibility.snapshot-id-inheritance.enabled | false    | Enables committing snapshots without explicit snapshot IDs; Not required for v2 tables    |
+
+### Catalog Properties
+
+| Property                          | Default            | Description                                            |
+| --------------------------------- | ------------------ | ------------------------------------------------------ |
+| catalog-impl                      | null               | a custom `Catalog` implementation to use by an engine  |
+| io-impl                           | null               | a custom `FileIO` implementation to use in a catalog   |
+| warehouse                         | null               | the root path of the data warehouse                    |
+| uri                               | null               | a URI string, such as Hive metastore URI               |
+| clients                           | 2                  | client pool size                                       |
+| cache-enabled                     | true               | Whether to cache catalog entries |
+| cache.expiration-interval-ms      | 30000              | How long catalog entries are locally cached, in milliseconds; 0 disables caching, negative values disable expiration |
+| lock-impl                         | null               | a custom implementation of the lock manager, the actual interface depends on the catalog used  |
+| lock.table                        | null               | an auxiliary table for locking, such as in [AWS DynamoDB lock manager](../aws/#dynamodb-for-commit-locking)  |
+| lock.acquire-interval-ms          | 5 seconds          | the interval to wait between each attempt to acquire a lock  |
+| lock.acquire-timeout-ms           | 3 minutes          | the maximum time to try acquiring a lock               |
+| lock.heartbeat-interval-ms        | 3 seconds          | the interval to wait between each heartbeat after acquiring a lock  |
+| lock.heartbeat-timeout-ms         | 15 seconds         | the maximum time without a heartbeat to consider a lock expired  |
+
+{{< hint info >}}
+`HadoopCatalog` and `HiveCatalog` can access the properties in their constructors.
+Any other custom catalog can access the properties by implementing `Catalog.initialize(catalogName, catalogProperties)`.
+The properties can be manually constructed or passed in from a compute engine like Spark or Flink.
+Spark uses its session properties as catalog properties, see more details in the [Spark configuration](../spark-configuration#catalog-configuration) section.
+Flink passes in catalog properties through `CREATE CATALOG` statement, see more details in the [Flink](../flink/#creating-catalogs-and-using-catalogs) section.
+{{< /hint >}}
+
+### Spark Catalog Configurations
```

Comment:
@samredai, when we talked about this before, I think the plan was to create a h2 section for Spark (and other engines) and to move the docs for how to use the read, write, and catalog properties, or at least have short explanations here. Can you add those things? I'd like to get this in since it's close.

Reply:
@rdblue I know this is close and has been open for a while, but as I add the instructions on read/write and the engine-specific properties, I feel that a Configuration section would be better than a single Configuration page. We really want to enable the "CMD+f" use-case for properties but that would always be at an engine specific level, right? Here's a rough diagram that I put together quickly as an example.
![iceberg-configuration-matrix](https://user-images.githubusercontent.com/43911210/180617873-49ffd559-6f67-4af9-ad49-223a89ac0972.png)
I'm curious to hear people's thoughts on:
- Configuration (drop-down section)
  - Spark Properties
  - Flink Properties
  - Hive Properties
  - Java Client Properties

We'd only need a section for engines where we provide a runtime as well as for the java client. Engines that provide native support always maintain the configuration docs on their own sites which we link to in the nav menu.

Reply:
@samredai, I think the current organization is close enough. We can think about moving things around later, but from the feedback I've heard we should have a single page with all of the configuration as a reference. Let's focus on getting that done first.

Reply:
@rdblue sounds good. I've updated the PR to include engine-specific sections with instructions on how to set the properties and included example code-snippets. The TOC looks like this now.
<img width="472" alt="Screen Shot 2022-08-13 at 4 19 06 AM" src="https://user-images.githubusercontent.com/43911210/184485231-39ff8cc1-b9ed-4681-8863-b215495f2056.png">

Since the table property names and descriptions are the same across all engines, I placed them in their own `Table Properties` section with sub-sections for read/write/behavior etc. and then link to them in the engine-specific table property sections that describe setting them. For example, here's the spark section.
<img width="934" alt="Screen Shot 2022-08-13 at 4 19 34 AM" src="https://user-images.githubusercontent.com/43911210/184486005-5de67808-c478-4a51-ad9c-0004aa307f02.png">

The catalog property names differ enough between engines that I explicitly list those in each engine-specific catalog section. Here's the spark catalog section.
<img width="895" alt="Screen Shot 2022-08-13 at 4 26 21 AM" src="https://user-images.githubusercontent.com/43911210/184487315-0f60ac7e-22a0-4500-9db1-7b23d9b6a6e0.png">